forge-game/src/main/java/forge/game/StaticEffect.java

```diff
@@ -252,6 +252,11 @@ final CardCollectionView remove() {
                 affectedCard.removeColor(getTimestamp(), ability.getId());
             }
 
+            // remove changed name
+            if (hasParam("SetName")) {
```

Comment:
If @Hanmac doesn't mind can you also check for `AddNames` there?

Seems Spy Kit currently doesn't reset because it's missing

Reply:
Great catch – just ran a test with `Spy Kit` and `Cornered Market` and things got ugly very quickly

I had wondered about "AddName" not being there, but forgot to test it... this is why you are the champion Bug Hunter